src/components/Plugin/ContributionRewardExtRewarders/Competition/Details.tsx

```diff
@@ -116,6 +116,9 @@ class CompetitionDetails extends React.Component<IProps, IStateProps> {
         this.setState({ showingSubmissionDetails: urlSubmission });
       }
     }
+
+    // Ensure the plugin's state is hydrated
+    await this.props.proposalState.plugin.entity.fetchState();
```

Comment:
Should this be in `withSubscription.createObservable`?

Reply:
Good point! Moved.